library/core/src/char/methods.rs

```diff
@@ -1116,6 +1117,24 @@ impl char {
         *self as u32 <= 0x7F
     }
 
+    /// Returns `Some` if the value is within the ASCII range,
+    /// or `None` if it's not.
+    ///
+    /// This is preferred to [`Self::is_ascii`] when you're passing the value
+    /// along to something else that can take [`ascii::Char`] rather than
+    /// needing to check again for itself whether the value is in ASCII.
+    #[must_use]
+    #[unstable(feature = "ascii_char", issue = "110998")]
+    #[inline]
+    pub const fn as_ascii(&self) -> Option<ascii::Char> {
```

Comment:
Why does this take `&self` and not `self`?

Reply:
I agree that `self` would make more sense, but I think that for the existing `is_ascii` above too.

So I made this `&self` for consistency with the existing methods here.  I'll add a note to the tracking issue to make sure that it's discussed further before stabilization.